Build failure "invalid conversion" in "utils/log.hpp"

I'm trying to compile SuperTuxKart on Alpine Linux. It uses Musl and GCC9. However I'm having an issue here I do not know how to resolve. While compiling I get the following:

```
/home/builder/aports/testing/supertuxkart/src/stk-code-1.0/src/utils/log.hpp: In static member function 'static void Log::error(const char*, const char*, ...)':
/home/builder/aports/testing/supertuxkart/src/stk-code-1.0/src/utils/log.hpp:103:48: error: invalid conversion from '__va_list_tag*' to 'int' [-fpermissive]
  103 |         printMessage(LEVEL, component, format, args);                \
      |                                                ^~~~
      |                                                |
      |                                                __va_list_tag*
```

--- a/src/utils/log.hpp
+++ b/src/utils/log.hpp
@@ -30,10 +30,10 @@
 #include <vector>
 
 
-#ifdef __GNUC__
+#if defined(__GLIBC__)
 #  define VALIST __gnuc_va_list
 #else
-#  define VALIST char*
+#  define VALIST va_list
 #endif
 
 #if defined(_WIN32) && defined(_MSC_VER) && _MSC_VER < 1800

